consultation_analyser/consultations/views/answers.py

```diff
@@ -144,98 +145,90 @@ def respondents_json(
     cache_key = f"respondents_{request.user.id}_{consultation_slug}_{question_slug}"
     cache_timeout = 60 * 20  #  20 mins
 
-    responseid = request.GET.get("responseid")
-    demographicindividual = request.GET.getlist("demographicindividual")
-    themesfilter = request.GET.getlist("themesfilter")
-    themesentiment = request.GET.get("themesentiment")
-    responsesentiment = request.GET.get("responsesentiment")
-    wordcount = request.GET.get("wordcount")
-
+    # Retrieve cached data
     data = cache.get(cache_key)
 
+    # If no cached data found
     if data is None:
-        # Get question data
-        consultation = get_object_or_404(models.Consultation, slug=consultation_slug)
-
-        question = models.Question.objects.get(
-            slug=question_slug,
-            consultation=consultation,
+        # Prefetch all related data to avoid multiple db hits
+        filtered_answers = models.Answer.objects.filter(
+            question_part__question__slug=question_slug
+        ).prefetch_related(
+            Prefetch("thememapping_set", to_attr="prefetched_thememappings")
```

Comment:
Ah cool - I didn't know about using the Prefetch class like this, so hadn't realised that we could get the `_set` references like this. Nice.

Reply:
I have also learnt something new about prefetch!